src/test/java/org/reaktivity/nukleus/maven/plugin/internal/GenerateMojoTest.java

```diff
@@ -15,76 +15,47 @@
  */
 package org.reaktivity.nukleus.maven.plugin.internal;
 
-import static org.junit.Assert.assertNotNull;
-
-import java.io.File;
-
 import org.antlr.v4.runtime.misc.ParseCancellationException;
-import org.apache.maven.plugin.testing.MojoRule;
-import org.apache.maven.project.MavenProject;
-import org.codehaus.plexus.configuration.PlexusConfiguration;
 import org.junit.Ignore;
 import org.junit.Rule;
 import org.junit.Test;
 
 public class GenerateMojoTest
 {
     @Rule
-    public MojoRule rule = new MojoRule();
+    public GenerateMojoRule rule = new GenerateMojoRule();
 
     @Test(expected = ParseCancellationException.class)
     public void shouldNotGenerateInvalidStructOctetsNotLast()
         throws Exception
     {
-        File pom = new File("src/test/resources/test-project/pom.xml");
-        MavenProject project = rule.readMavenProject(new File("src/test/resources/test-project"));
-        GenerateMojo myMojo = (GenerateMojo) rule.lookupConfiguredMojo(project, "generate");
-        assertNotNull(myMojo);
-
-        PlexusConfiguration configuration = rule.extractPluginConfiguration("nukleus-maven-plugin", pom);
-        configuration.addChild("scopeNames", "invalidOctetsNotLast");
-        configuration.addChild("packageName", "org.reaktivity.reaktor.internal.test.types");
-        configuration.addChild("inputDirectory", "src/test/resources/test-project");
-        configuration.addChild("outputDirectory", "target/generated-flyweights");
-        rule.configureMojo(myMojo, configuration);
-        myMojo.execute();
+        rule.scopeNames("invalidOctetsNotLast")
+            .packageName("org.reaktivity.reaktor.internal.test.types")
+            .inputDirectory("src/test/resources/test-project")
+            .outputDirectory("target/generated-test-sources/test-reaktivity")
```

Comment:
Move `packageName(...)`, `inputDirectory(...)` and `outputDirectory(...)` to rule declaration, leaving just `scopeNames(...)` and `execute()` in the test body.

Reply:
done